File: Java/reverse-string.java
/*
* https://leetcode.com/problems/reverse-string/
* Runtime: 1 ms, faster than 99.25% of Java online submissions for Reverse String.
* Memory Usage: 42.9 MB, less than 99.41% of Java online submissions for Reverse String.
*/

class Solution {
    public void reverseString(char[] s) {
        int j = s.length - 1;
        char temp;
        for (int i = 0; i < j; i++) {
            temp = s[i];
            s[i] = s[j];
            s[j] = temp;
            j--;
        }
    }
}

File: C++/reverse-string.cpp
/*
* https://leetcode.com/problems/reverse-string/
* Runtime: 36 ms, faster than 99.55% of C++ online submissions for Reverse String.
* Memory Usage: 15.2 MB, less than 95.12% of C++ online submissions for Reverse String.
*/

class Solution {
public:
    void reverseString(vector<char>& s) {
        int j = s.size() - 1;        
        for (int i = 0; i < j; i++) {
            char temp = s[i];
            s[i] = s[j];
            s[j--] = temp;
        }
    }
};
